src/D2L.CodeStyle.Analyzers/ApiUsage/ContentPhysicalPaths/PhysicalPathPropertyAnalysis.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System.Collections.Immutable;
+using System.Linq;
+using D2L.CodeStyle.Analyzers.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace D2L.CodeStyle.Analyzers.ApiUsage.ContentPhysicalPaths {
+
+	internal sealed class PhysicalPathPropertyAnalysis {
+
+		internal static readonly DiagnosticDescriptor DiagnosticDescriptor = Diagnostics.ContentPhysicalPathUsages;

```

Comment:
nit: I wouldn't be offended if you just inlined this (I was reading `ReportDiagnostic`, saw that the descriptor was passed down, looked up each caller and saw that it was coming from this property, and then had to find this property to figure out which diagnostic it was... not sure it helps to hide that?)